kernel/reduction.mli

```diff
@@ -1,25 +1,33 @@
 (** Term reduction and conversion test. *)
-
 open Basic
 open Term
 
 type Debug.flag += D_reduce
 
-type red_strategy = Hnf | Snf | Whnf
+type red_target   = Snf | Whnf
+type red_strategy = ByName | ByValue | ByStrongValue
 
 type red_cfg = {
-  select : (Rule.rule_name -> bool) option;
+  select   : (Rule.rule_name -> bool) option;
   nb_steps : int option; (* [Some 0] for no evaluation, [None] for no bound *)
-  strategy : red_strategy;
-  beta : bool
+  target   : red_target;
+  strat    : red_strategy;
+  beta     : bool;
+  logger   : position -> Rule.rule_name -> term Lazy.t -> unit
 }
+(** Configuration for reduction.
+    [select] = [Some f] restreins rules according to the given filter on names.
+    [select] = [None] is the default behaviour (all rules allowed).
+    [nb_steps] = [Some n] Allows only [n] reduction steps.
+    [nb_steps] = [None] is the default behaviour.
+    [target] is the normal form to compute.
+    [strat] is the reduction strategy.
+    [beta] flag enables/disables beta reductions.
+    [logger] is the function to call upon applying a reduction rule.
```

Comment:
Default logger is `fun _ _ _ -> ()`. I think it's simple enough that people can create it themselves. Otherwise they can still use `default_cfg.logger`.

Other loggers can easily be created by the user. For instance, logging how many time the rule `r` is used is simply done with : `fun _ rn _ -> if rule_name_eq rn r then incr counter`